google-cloud-storage/src/test/java/com/google/cloud/storage/ParallelCompositeUploadWritableByteChannelTest.java

```diff
@@ -749,6 +766,49 @@ public BlobInfo internalObjectGet(BlobId blobId, Opts<ObjectSourceOpt> opts) {
         () -> assertThat(storageInternal.deleteRequests).containsExactly(p1, p2, p3));
   }
 
+  @Test
+  public void partMetadataFieldDecoratorUsesCustomTime() throws IOException {
+    TestClock clock = TestClock.tickBy(Instant.EPOCH, Duration.ofSeconds(1));
+    OffsetDateTime rangeBegin =
+        OffsetDateTime.from(Instant.EPOCH.plus(Duration.ofSeconds(29)).atZone(ZoneId.of("Z")));
+    OffsetDateTime rangeEnd =
+        OffsetDateTime.from(Instant.EPOCH.plus(Duration.ofMinutes(2)).atZone(ZoneId.of("Z")));
+
+    FakeStorageInternal storageInternal =
+        new FakeStorageInternal() {
+          @Override
+          public BlobInfo internalDirectUpload(
+              BlobInfo info, Opts<ObjectTargetOpt> opts, ByteBuffer buf) {
+            if (info.getBlobId().getName().endsWith(".part")) {
+              // Kinda hacky but since we are creating multiple parts we will use a range
+              // to ensure the customTimes are being calculated appropriately
+              assertThat(info.getCustomTimeOffsetDateTime().isAfter(rangeBegin)).isTrue();
+              assertThat(info.getCustomTimeOffsetDateTime().isBefore(rangeEnd)).isTrue();
+            } else {
+              assertThat(info.getCustomTimeOffsetDateTime()).isNull();
+            }
+            return super.internalDirectUpload(info, opts, buf);
+          }
+        };
+    ParallelCompositeUploadWritableByteChannel pcu =
+        new ParallelCompositeUploadWritableByteChannel(
+            bufferHandlePool,
+            MoreExecutors.directExecutor(),
+            partNamingStrategy,
+            PartCleanupStrategy.always(),
+            10,
+            PartMetadataFieldDecorator.setCustomTimeInFuture(Duration.ofSeconds(30))
+                .newInstance(clock),
+            finalObject,
+            storageInternal,
+            info,
+            opts);
+    byte[] bytes = DataGenerator.base64Characters().genBytes(bufferCapacity * 3 - 1);
+    pcu.write(ByteBuffer.wrap(bytes));
+
+    pcu.close();
+  }
+
```

Comment:
Can we add some unit tests for `CustomTimeInFuture` itself and to simplify verifying it's time math and timezone retention.